src/Plugin/Journal.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Http\Client\Common\Plugin;
+
+use Http\Client\Exception;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Records history of HTTP calls.
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+interface Journal
```

Comment:
The reason I suggested message is that it is mostly message related. Out of four/three dependencies three/two are messages. Implementations will probably never be in any of our repositories, because (as you said) they are mostly framework specific.

Maybe we could move the Journal together with the history plugin to it's own repository? It depends if we find any use cases for Journal apart from History plugin. Since it is relatively simple, I would say we don't necessarily need to think about any scenario.